corehq/apps/locations/models.py

```diff
@@ -349,7 +336,7 @@ def filter_path_by_user_input(self, domain, user_input):
             Massachusetts/Middlesex/Cambridge
         It matches by name or site-code
         """
-        direct_matches = self.filter_by_user_input(domain, user_input)
+        direct_matches = self._user_input_filter(domain, user_input)
         return self.get_queryset_descendants(direct_matches, include_self=True)
```

Comment:
Do both implementations of `get_queryset_descendants` work on `Q` objects as well as querysets?

Reply:
Oh, I see you modified them to allow that.  Makes sense.